pkg/install/deploystorage.go

```diff
@@ -103,12 +104,28 @@ func (i *Installer) deployStorageTemplate(ctx context.Context, installConfig *in
 
 		applications := graphrbac.NewApplicationsClient(spp.TenantID, spGraphAuthorizer)
 
-		res, err := applications.GetServicePrincipalsIDByAppID(ctx, spp.ClientID)
+		timeoutCtx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+		defer cancel()
+		// NOTE: Do not override err with the error returned by wait.PollImmediateUntil.
+		// Doing this will not propagate the latest error to the user in case when wait exceeds the timeout
```

Comment:
I think we should phrase this as an explanation of the existing code rather than as an instruction to not change the code in a certain way, e.g.

> Errors returned from `PollImmediateUntil` are discarded so that they do not overwrite `Authorization_IdentityNotFound` errors which may occur within the `ConditionFunc` passed to `PollImmediateUntil`.

Reply:
am fine with this for now given time constraints